src/features/AaveTokenMigration/helpers.ts

```diff
@@ -0,0 +1,32 @@
+import { DEFAULT_ASSET_DECIMAL, DEFAULT_NETWORK_CHAINID } from '@config';
+import { AaveMigrator } from '@services/EthService/contracts';
+import { ITokenMigrationFormFull, ITxData, ITxObject, ITxToAddress } from '@types';
+import { formatApproveTx, inputGasPriceToHex, inputValueToHex, toWei } from '@utils';
+
+import { MIGRATION_CONTRACT } from './config';
+
+export const createApproveTx = (
+  payload: ITokenMigrationFormFull
+): Omit<ITxObject, 'nonce' | 'gasLimit'> =>
+  formatApproveTx({
+    contractAddress: payload.asset.contractAddress as ITxToAddress,
+    baseTokenAmount: toWei(payload.amount, DEFAULT_ASSET_DECIMAL),
+    fromAddress: payload.account.address,
+    spenderAddress: MIGRATION_CONTRACT,
+    chainId: DEFAULT_NETWORK_CHAINID,
+    hexGasPrice: inputGasPriceToHex(payload.gasPrice)
+  });
+
+export const createMigrationTx = (
+  payload: ITokenMigrationFormFull
+): Omit<ITxObject, 'nonce' | 'gasLimit'> => {
+  const data = AaveMigrator.migrateFromLEND.encodeInput({ amount: payload.amount });
```

Comment:
I'm assuming this is where the wrong amount came in. Here was my test approve tx: https://etherscan.io/tx/0x5ecb4ea41154b1a8f6cc3cb8af5f7162f33cd59c92708b337fc69efd67310e17
And my test migrate tx: https://etherscan.io/tx/0x4be2459a0fe228764f675b337d8228452d2a04e448cb32c166b42ba8327d180e